### ISSUE TITLE
Analyzer suggest incorrect UpperCamelCase for structs with keyword name

### ISSUE THREAD

**Issue by joellelard** [rust-lang/rust-analyzer#23304]
While looking at rust documentation, I discovered that you could name things after keywords by adding `r#` as prefix, ex: `let r#pub: i32 = 5;`. Very suprised, I tried to do it (and of course it worked), but when trying it with a struct name, like this:
```rust
struct r#pub;
```
It didn't throw any errors, but rust-analyzer complained saying that I should use UpperCamelCase:
```
Structure `r#pub` should have UpperCamelCase name, e.g. `R#pub`
```
But that's incorrect, it should suggest either `Pub` or `r#Pub`, but not `R#pub`. And when trying to compile, the compiler gives a correct warning of:
```
struct r#pub;
       ^^^^^ help: convert the identifier to upper camel case: `Pub`
```

Also, I'm sorry but I'm using rust-analyzer on nvf (neovim configured with nix), and I don't know how to check its version. But I did try to check if there were any similar issues and didn't see any, closed or otherwise.

If this is not reproducible, I will try to dig around to find the versions of all the things I used.
